app/tool_proxy.rb

```diff
@@ -37,7 +39,7 @@ def as_json
   # Generates a 128 hexadecimal character string for the Tool Provider's
   # half of the shared secret (See section 5.6).
   def tp_half_shared_secret
-    @_tp_half_shared_secret ||= SecureRandom.hex(64)
+    @tp_half_shared_secret ||= SecureRandom.hex(64)
```

Comment:
It seems like there will some bit of oddity with this since tp_half_shared_secret is a field on this model. Naming instance variables and methods the same thing makes my head spin a little. Maybe @westonkd knows why we had the underscore in front originally?

Reply:
in that case maybe it should be `self.tp_half_shared_secret ||= SecureRandom.hex(64)` ?

Reply:
wait a minute, `tp_half_shared_secret` isn't actually an attribute on the model that is saved in the database...

Reply:
My bad. I assumed that it was because of the validator.